### PR TITLE
Migrate to Flask

### PR DESCRIPTION
**Pull request by Ap0c**
## Why?

More widely supported than Bottle with more features and plugins, whilst at the same time not requiring an app structure that differs too much from our current one at the moment. More specifically, we'd like to use Jinja2 because it's a *lot* more powerful than the template engine that comes with Bottle. Also, while it's true that you can use Jinja2 with Bottle, it's better integrated with Flask.

## Changes

- Added a README with install and run instructions.
- Updated requirements with Flask, and bumped requests.
- Added run script with `DEBUG` turned on for the moment.
- Migrated templates to Jinja2.
- Moved assets and templates out into main dir temporarily, until we can set up a package structure (possibly?).
- Migrated main site module over to Flask from Bottle.
- Dropped `__pycache__/` from .gitignore because I *think* it should be covered by the `*.pyc` rule.

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export FLASK_APP=views.py
```
Comment (hmac) [inline run.sh:1]: what does this env var do?

**Review by hmac — APPROVED** [review]
:+1: :+1:

**Comment by Ap0c** [comment]
Whoops, didn't notice `requests` had been bumped in #3.

File: run.sh (hunk 1 above)
Comment (Ap0c) [reply run.sh:1]: That's so the `flask` command knows where you've put the application instance ([docs](http://flask.pocoo.org/docs/0.12/quickstart/#a-minimal-application)).

**Pull request merged**